pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -81,27 +83,41 @@ func PodChanged(oldPod, newPod *v1.Pod, endpointChanged EndpointsMatch) (bool, b
 
 // GetPodServiceMemberships returns a set of Service keys for Services that have
 // a selector matching the given pod.
-func GetPodServiceMemberships(serviceLister v1listers.ServiceLister, pod *v1.Pod) (sets.String, error) {
+func GetPodServiceMemberships(serviceLister v1listers.ServiceLister, selectorCache *sync.Map, pod *v1.Pod) (sets.String, error) {
 	set := sets.String{}
-	services, err := serviceLister.GetPodServices(pod)
+	services, err := serviceLister.Services(pod.Namespace).List(labels.Everything())
 	if err != nil {
 		// don't log this error because this function makes pointless
 		// errors when no services match
-		return set, nil
-	}
-	for i := range services {
-		key, err := controller.KeyFunc(services[i])
-		if err != nil {
-			return nil, err
+		return set, err
+	}
+
+	var selector labels.Selector
+	for _, service := range services {
+		if v, ok := selectorCache.Load(fmt.Sprintf("%s_%s", service.Namespace, service.Name)); ok {
+			selector = v.(labels.Selector)
+		} else {
+			if service.Spec.Selector == nil {
+				// services with nil selectors match nothing, not everything.
```

Comment:
this comment is a bit confusing.
please reword it to clarify the intent of the `continue` - e.g. `if the service has a nil selector this means that ....`